### PR TITLE
Functions added for calling api endpoints

### PR DESCRIPTION
**Pull request by vintydremio**

### CODE REVIEW

File: dremio_client/model/endpoints.py [hunk 1]
```diff
@@ -258,9 +264,45 @@ def create_user(token, base_url, json, ssl_verify=True):
     return _post(base_url + "/api/v3/user", token, json, ssl_verify=ssl_verify)
 
 
+#def get_user(token, base_url, uid, ssl_verify=True):
```
Comment (rymurr) [inline dremio_client/model/endpoints.py]: please remove the dead code

File: dremio_client/model/endpoints.py [hunk 2]
```diff
@@ -284,6 +326,70 @@ def user(token, base_url, uid=None, name=None, ssl_verify=True):
...
+
+def delete_role(token, base_url, uid , ssl_verify=True):
+    return _delete(base_url + "/api/v3/role/{}".format(uid) , token, ssl_verify=ssl_verify)
+
+
+def update_role(token, base_url, uid, json, ssl_verify=True):
+    return _put(base_url + "/api/v3/role/{}".format(uid), token, json, ssl_verify=ssl_verify)
+
+def get_all_roles(token, base_url ,startIndex=None , count=None ,query=None  , ssl_verify=True):
+    if startIndex is not None and count is not None and query is not None:
```
Comment (rymurr) [inline dremio_client/model/endpoints.py]: this (and the ones below it) can be simplified.
May I suggest somethign like (i didn't test this):
```
params = dict()
if startIndex is not None:
  params['startIndex'] = startIndex
...repeat for other params
query = '&'.join("{}={}".format(k,v) for k,v in params)
return _get(base_url + "/api/v3/role{}".format(("?" + query) if query else "" )
```

File: dremio_client/dremio_simple_client.py [hunk 3]
```diff
@@ -171,9 +174,33 @@ def votes(self):
...
+    #def get_user(self, uid):
+     #   return get_user(self._token, self._base_url, uid, ssl_verify=self._ssl_verify)
+
+    def delete_user(self, uid):
+        return delete_user(self._token, self._base_url, uid, ssl_verify=self._ssl_verify)
+
     def update_user(self, uid, json):
         return update_user(self._token, self._base_url, uid, json, ssl_verify=self._ssl_verify)
 
+    def get_all_users(self, startIndex=None , count=None ,query=None ):
```
Comment (rymurr) [inline dremio_client/dremio_simple_client.py:203]: thanks for including pydoc! Can you add the same for the rest of the methods you added?

**Review by rymurr — CHANGES_REQUESTED** [review]

**Comment by vintydremio** [comment]
Please review the changes made @rymurr

File: dremio_client/model/endpoints.py [hunk 4]
```diff
@@ -255,12 +261,88 @@ def votes(token, base_url, ssl_verify=True):
...
+    params = dict()
+    if startIndex is not None:
+        params['startIndex'] = startIndex
+    if count is not None:
+        params['count'] = count
+    if query is not None:
+        params['query'] = query
+    end_url = base_url + "/api/v3/user"
+    qry = '&'.join("{}={}".format(k, v) for k, v in params.items())
+    if len(qry)>0:
```
Comment (rymurr) [inline dremio_client/model/endpoints.py]: it should be sufficient to do `if query:`

File: dremio_client/model/endpoints.py [hunk 5]
```diff
@@ -284,6 +366,142 @@ def user(token, base_url, uid=None, name=None, ssl_verify=True):
...
+    :return: result object
+    """
+    params = dict()
+    if startIndex is not None:
+        params['startIndex'] = startIndex
+    if count is not None:
+        params['count'] = count
+    qry = '&'.join("{}={}".format(k, v) for k, v in params.items())
+    end_url = base_url + "/api/v3/role/{}/privilege".format(rid)
+    if len(qry) > 0:
```
Comment (rymurr) [inline dremio_client/model/endpoints.py]: There is a lot of duplication in these methods. May I suggest something like:

``` python
def build_url(**kwargs):
    query = "&".join("{}={}".format(k,v) for k,v in kwargs.items() if v)
    return "?{}".format(url, query) if query else ""

# replace in methods with
end_url = base_url + "/api/v3/role/{}/privilege".format(rid) + build_url(start_index=start_index, count=count)
```

**Review by rymurr — CHANGES_REQUESTED** [review]

File: dremio_client/model/endpoints.py (hunk 5 above)
Comment (vintydremio) [reply dremio_client/model/endpoints.py]: code has been simplified (duplication removed) @rymurr please review

**Review by rymurr — APPROVED** [review]

**Pull request merged**